[flexibo] iP

flexibo opened the pull request:
# **Qlatz To-Do List**
## Qlatz To-Do List
### Qlatz To-Do List
#### Qlatz To-Do List
##### Qlatz To-Do List
###### Qlatz To-Do List
> _I like boogers_ 

- one booger
- two boogers
- theree boogers [typewriter](https://youtu.be/Bs5TEuZPQl8?si=3X4Rqm9dFkQcVEyB)

> [!WARNING]
> May cause bleedng
``nosepickercolorpicker`` `#c900c3` 🤓 
1. pick finger
2. dont cut nails, or wash them for 1 week
3. jam it in nose


- [x] #466
- [ ] Add delight to the experience when all tasks are complete :tada:
- [x] https://github.com/ikatyang/emoji-cheat-sheet/blob/master/README.md
- [ ] test
- [x] test

```java 
 System.out.println("Working Directory = " + System.getProperty("user.dir"));
        
 File f = new File("src/main/resources/image0.jpg");
 if(f.exists() && !f.isDirectory()) { 
      System.out.println("I AM ALIVE");
        // do something
 } else {
      crash();
 }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
Level-0 created greetings

Commit:
Level-0 created greetings

Commit:
Level-1 Echo

Commit:
Level-2 Add, List

Commit:
Level-3 Mark as Done

Commit:
Level 4 - ToDos, Events, Deadlines

Commit:
Level 4 - ToDos, Events, Deadlines

Commit:
A-TextUiTesting: Automated Text UI Testing

Commit:
Level 5 - Handle Errors

Commit:
Level 5 - Handle Errors

Commit:
Level 6 - Delete

Commit:
A-Enums

Commit:
A-Enums

Commit:
A-Enums

Commit:
Level 7 Save, add to brranch levl 77777777777777777777777

Commit:
Level 7 Save, add to brranch levl 77777777777777777777777

Commit:
Level 7 Save, add to brranch levl 77777777777777777777777

Commit:
Merge branch 'branch-level-7'

Commit:
lelve 8

Commit:
Merge branch 'branch-level-8'

Commit:
Change code format to be more OOP friendly.

Commit:
Merge branch 'branch-a-moreoop'

Commit:
Organized files to standard packages.

Created new packages for "exception", "task", and "command" for enhanced readability.

Commit:
Removed Duke Welcome.

Commit:
Merge branch 'add-gradle-support'

Commit:
Fixed file not creating on corruption bug.

Commit:
Add test cases for Add and Delete Command classes.

Commit:
Created JAR file

Commit:
Added JavaDocs.

Commit:
Added JavaDocs.

Added for the rest of the files except for the test files.

Commit:
Minor edit to follow coding standard.

Commit:
Added Find command and function.

Commit:
Merge branch 'branch-a-codingstandard'

Commit:
Minor edit to fit coding standard.

Commit:
Merge branch 'branch-a-codingstandard'

Commit:
Minor comment change to JavaDocs.

Commit:
Merge branch 'branch-a-javadoc'

Commit:
Minor comment change to JavaDocs.

Commit:
Implement checkstyle, edit import statements order.

Commit:
Merge branch 'branch-a-checkstyle'

victorlaiyeeteng commented on src/main/java/duke/main/TaskList.java:
```diff
@@ -0,0 +1,109 @@
+        tasks.get(i).unmark();
+    }
+
+    public ArrayList<Task> getTasks() {
+        return tasks;
+    }
```
Does this need a header comment?

flexibo replied:
Hi! If I'm not wrong, the setter and getter functions do not need the any comments
![image](https://github.com/nus-cs2103-AY2324S1/ip/assets/40513197/d99e51d4-452f-4cea-b7d5-9fa65a42b500)

victorlaiyeeteng commented on src/main/java/duke/main/TaskList.java:
```diff
@@ -0,0 +1,109 @@
+     * @param searchString String to search with.
+     */
+    public void print(String searchString) {
+        boolean found = false;
```
```suggestion
        boolean isFound = false;
```

victorlaiyeeteng commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,28 @@
+     * @param task Task.
+     * @param dateTime Deadline for task.
+     */
+    public Deadline(@JsonProperty("task") String task, @JsonProperty("dateTime") LocalDate dateTime) {
```
I like the use of `@JsonProperty` in the constructor.

victorlaiyeeteng commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,164 @@
+package duke.main;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import duke.command.AddCommand;
+import duke.command.ChangeMarkCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.exception.InvalidCommandException;
+import duke.exception.InvalidDateFormatException;
+import duke.exception.InvalidParametersException;
+import duke.exception.MissingCommandException;
+import duke.exception.MissingParametersException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ToDo;
```
I like the good practice of ordering the imported packages well based on lexicographical order.

flexibo replied:
Intellij auto sorter is pog.

victorlaiyeeteng commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,164 @@
+
+/** Deals with making sense of user command, manages error handling based on user input */
+public class Parser {
+    private static String[] splitText;
```
Is there another way to name `splitText` as it may sound to the reviewer as a verb. Perhaps `formattedText`?

flexibo replied:
Hmmm, sounds good, but I would like to make this variable obvious that it contains the split form of the string. I initially thought of ``splittedText`` instead of ``splitText`` but then realized that the past tense of split is split 🤔.

victorlaiyeeteng reviewed (commented):
The coding standards look really good. I suggested a few improvements on naming of variables and some other suggestions.

flexibo committed:
Update TaskList.java change found variable.

Change boolean from "found" to "isFound" according to suggestion.

Co-authored-by: Victor Lai <75977350+victorlaiyeeteng@users.noreply.github.com>

jflam1301 commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,164 @@
+     */
+    private static LocalDate getDateWithCommand(String str, String command) throws MissingCommandException,
+            MissingParametersException, InvalidDateFormatException {
+        boolean found = false;
```
Maybe the name of the boolean can be isFound?

flexibo replied:
Yup done from previous comment, thanks!

jflam1301 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,50 @@
+    @JsonSubTypes.Type(value = Deadline.class, name = "Deadline")
+})
+public class Task {
+    private boolean done = false;
```
Maybe change the name of the boolean to isDone?

jflam1301 commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,31 @@
+
+/** Command to find tasks based on search string */
+public class FindCommand implements Command {
+    private final String searchString;
```
searchString sounds like a verb, maybe can name it to something like stringToSearch? (although searchString still sounds better)

flexibo replied:
hmm yeah, it sounds a bit odd, but I will change it to stringToSearch just to be safe :D

jflam1301 commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,164 @@
+        // Split text into two.
+        // With index 0 be the first word of user input, and index 1 containing the rest of the string.
+        inputString = inputString.toLowerCase();
+        splitText = inputString.split(" ", 2);
```
Maybe you can put this.splitText to distinguish the local and global variables since they are put next to each other this time?

flexibo replied:
apologies, can you elaborate on this? I'm not very sure which variable is considered to be global in this case.

jflam1301 replied:
sorry for the confusion, since splitText is a member of the class and inputString is a local variable to this function, maybe you can use this.splitText to specify that it is a member of the class because I was confused for a second thinking whether it is a local variable or not.

jflam1301 reviewed (commented):
Overall, I enjoying reviewing your code because they are very neat and structured. Keep up the good work!

Commit:
Implement GUI, Fix JSON bug from prev commit.

Commit:
Merge branch 'branch-level-10'

Commit:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/duke/main/TaskList.java

Commit:
Implement VarArgs for find command.

Commit:
Merge branch 'branch-a-varargs'

Commit:
Main classes: Update to follow code quality

Increase readability of code.

Change the ordering of if statements to make happy path more obvious,
and adjust code to avoid arrowhead.

Refer to this for more info on code quality
https://nus-cs2103-ay2324s1.github.io/website/se-book-adapted/chapters/codeQuality.html#code-quality

Commit:
Command interface: Remove semicolon

Commit:
TaskList: Add assertion for adding task

Check if the task being added is added to the back of the list as stated
in the JavaDocs.

Commit:
Find command: Change search method to use streams

Search method use streams instead of a nested for loop.

Make it faster to find and filter tasks with large number of tasks.

flexibo committed:
Merge pull request #2 from flexibo/branch-A-Streams

Find command: Change search method to use streams

flexibo committed:
Merge branch 'master' into branch-A-CodeQuality

flexibo committed:
Merge pull request #3 from flexibo/branch-A-CodeQuality

Branch a code quality

flexibo committed:
Merge pull request #4 from flexibo/branch-A-Assertions

TaskList: Add assertion for adding task

Commit:
Add Continuous Integration

Reference from:
https://github.com/se-edu/duke/blob/full-template/.github/workflows/gradle.yml

Commit:
Task Class: Add Tags to Task

Add 2 new commands, "Tag" and "Tags" for users to add and view tags of task.

Easier for users to sort or organize tasks.

Commit:
Imports: Removed unused imports

Commit:
Merge branch 'branch-c-tagging'

Commit:
GUI: Update GUI and add sound

Add sound each time the user press on enter,
add 3 new commands, play, pause, and stop for music.

Commit:
Cleanup some warnings

Commit:
Add representative screenshot of the product

Commit:
Update representative screenshot of the product

The top message was not shown fully in the
previous version.

Commit:
Update representative screenshot of the product

The top message was not shown fully in the
previous version.

Commit:
Update representative screenshot of the product

The top message was not shown fully in the
previous version.

Commit:
Music related commands: Add JavaDocs

Commit:
CheckStyle: Edited checkstyle errors accordingly

flexibo committed:
Update README.md

Commit:
Parser: Update formatting issues

Commit:
Parser: Update format issues

Commit:
Test: Add test case for Add, Delete, Change mark

Commit:
JSON: Fixed mark task bugs

Commit:
Remove unused import statements

Commit:
Update Ui.png

Commit:
Update Ui.png

Commit:
Fix Checkstyle Error in Parser.java

Commit:
Bug fix: Unable to show app on Mac

Commit:
Remove warnings from .fxml files

Commit:
Remove warnings from .fxml files

Commit:
Improve GUI and add bye command

Commit:
Improve GUI and add bye command

Commit:
Fix scroll issue

Commit:
Fix scroll issue

Commit:
Update scroll look
